### PR TITLE
feat: [IA-241] Implements new Payment Transaction error screen

### PR DESCRIPTION
**Pull request by CrisTofani**
## Short description
This PR reworks the Transaction Error screen shown after the activation of a payment in case of errors. This PR handles the errors implemented by the backend in pagopa/io-backend#828

## How to test
Change the error response on `io-dev-api-server` [here](https://github.com/pagopa/io-dev-api-server/blob/master/src/routers/payment.ts#L44) or [here](https://github.com/pagopa/io-dev-api-server/blob/master/src/routers/payment.ts#L63) to return the newly defined error payload.

[Here](https://docs.google.com/spreadsheets/d/16VBHabKr4b3HBII-P0VuLByhWLRSU4Y3-TwYMu077MU/edit#gid=0) the map of the complete cases of errors given by the new implementation

| Error Type | Screen |
| ------ | ------- |
| **TECHNICAL** |  |
| **DATA** |  |
| **EC** |  | 
| **ONGOING** |  | 
| **EXPIRED** |  |
| **REVOKED** |  |
| **DUPLICATED** |  |
| **UNCOVERED** |  |

### CODE REVIEW

**Comment by thisisjp** [comment]
@CrisTofani before reviewing the copy, please note there shouldn't be any header (nor back button neither title) in this page

**Comment by thisisjp** [comment]
@CrisTofani sorry for the double comment: in the umbrella screen the description seems aligned on the left, while it should be centered—is it just a copy-paste issue?

**Comment by CrisTofani** [comment]
> @CrisTofani before reviewing the copy, please note there shouldn't be any header (nor back button neither title) in this page

Fixed within 7cc35fb

> @CrisTofani sorry for the double comment: in the umbrella screen the description seems aligned on the left, while it should be centered—is it just a copy-paste issue?

It was a style issue on the code, thank's for catching it! Fixed within 79a3c57

**Comment by Undermaken** [comment]
@CrisTofani I added the label ✋🏻  `don't merge` until [this PR](https://github.com/pagopa/io-backend/pull/828) is not merged and deployed
btw we can review this PR

**Comment by CrisTofani** [comment]
> @CrisTofani I added the label ✋🏻 `don't merge` until [this PR](https://github.com/pagopa/io-backend/pull/828) is not merged and deployed
> btw we can review this PR

Yes, thanks! I'm waiting for news by the backend team, once this will be merged I will remove the tag

File: locales/en/index.yml
Comment (thisisjp) [inline locales/en/index.yml:1376]:
```suggestion
    TECHNICAL: Sorry, a technical problem has occurred with this notice
```

File: locales/en/index.yml
Comment (thisisjp) [inline locales/en/index.yml:1377]:
```suggestion
    DATA: Sorry, the payment information is incorrect
```

File: locales/en/index.yml
Comment (thisisjp) [inline locales/en/index.yml:1380]:
```suggestion
    ONGOING_SUBTITLE: If it has been too long, please let us know!
```

File: locales/en/index.yml
Comment (thisisjp) [inline locales/en/index.yml:1381]:
```suggestion
    REVOKED: Sorry, the Payee has revoked this notice
```

File: locales/en/index.yml
Comment (thisisjp) [inline locales/en/index.yml:1382]:
```suggestion
    EXPIRED: Sorry, the notice has expired, it is not possible to pay it
```

File: locales/en/index.yml
Comment (thisisjp) [inline locales/en/index.yml:1383]:
```suggestion
    GENERIC_ERROR: Sorry, an unexpected error has occurred
```

File: locales/en/index.yml
Comment (thisisjp) [inline locales/en/index.yml:1384]:
```suggestion
    GENERIC_ERROR_SUBTITLE: You can try again, or contact us to get support.
```

File: locales/en/index.yml
Comment (thisisjp) [inline locales/en/index.yml:1385]:
```suggestion
    PAYMENT_DUPLICATED: This notice has been already paid!
```

File: locales/en/index.yml
Comment (thisisjp) [inline locales/en/index.yml:1396]:
```suggestion
    contactECsubtitle: Contact the Institution for more information.
```

File: locales/en/index.yml
Comment (thisisjp) [inline locales/en/index.yml:1397]:
```suggestion
    assistanceLabel: "Use this code to get support:"
```

File: locales/it/index.yml
Comment (thisisjp) [inline locales/it/index.yml:1400]:
```suggestion
    DATA: Spiacenti, i dati dell’avviso non sono corretti
```

File: locales/it/index.yml
Comment (thisisjp) [inline locales/it/index.yml:1402]:
```suggestion
    PAYMENT_ONGOING: Il pagamento è già in corso, riprova tra qualche minuto
```

File: locales/it/index.yml
Comment (thisisjp) [inline locales/it/index.yml:1407]:
```suggestion
    GENERIC_ERROR_SUBTITLE: Prova di nuovo o contattaci per ricevere assistenza.
```

File: locales/it/index.yml
Comment (thisisjp) [inline locales/it/index.yml:1420]:
```suggestion
    assistanceLabel: "Per ricevere assistenza usa il codice:"
```

**Review by thisisjp — CHANGES_REQUESTED** [review]

**Review by Undermaken — CHANGES_REQUESTED** [review]

File: ts/screens/wallet/payment/TransactionErrorScreen.tsx
Comment (CrisTofani) [reply ts/screens/wallet/payment/TransactionErrorScreen.tsx:302]: No contextual help or back button is needed by design since the user can access to assistance by tapping the related CTA on the footer

File: ts/screens/wallet/payment/TransactionErrorScreen.tsx
Comment (CrisTofani) [reply ts/screens/wallet/payment/TransactionErrorScreen.tsx:65]: This was on my TODO list, and then I forgot to add it, thanks for catching!

File: ts/utils/payment.ts
Comment (CrisTofani) [reply ts/utils/payment.ts:288]: In the beginning I was thinking about the same refactoring then noticed that these localized labels was even translated in Deutsch language so I preferred not to rename them, anyway thinking it better now we can for sure refactorize it

File: ts/screens/wallet/payment/TransactionErrorScreen.tsx
Comment (CrisTofani) [reply ts/screens/wallet/payment/TransactionErrorScreen.tsx:99]: yes this is a const, but actually in this case is scoped only to be used by the current function, why to move on the entire file?

File: ts/screens/wallet/payment/TransactionErrorScreen.tsx
Comment (Undermaken) [inline ts/screens/wallet/payment/TransactionErrorScreen.tsx:73]: Since it is used twice, could be a refactoring place it in the util file `utils/payment` and share between the components?

File: ts/screens/wallet/payment/TransactionErrorScreen.tsx
Comment (Undermaken) [reply ts/screens/wallet/payment/TransactionErrorScreen.tsx:124]: because every time the function is called the const is created

File: ts/screens/wallet/payment/TransactionErrorScreen.tsx
Comment (Undermaken) [reply ts/screens/wallet/payment/TransactionErrorScreen.tsx:6]: @CrisTofani 
Did you see this refactoring proposal? what do you think about it, could it be valid?

**Review by Undermaken — CHANGES_REQUESTED** [review]

File: ts/screens/wallet/payment/TransactionErrorScreen.tsx
Comment (CrisTofani) [reply ts/screens/wallet/payment/TransactionErrorScreen.tsx:6]: @Undermaken I missed it through all the messages 😅 

The refactoring looks good, just see one POF (talking about ui):

this code
```
  const subtitle = validError.fold(
    _ => (
      <H4 weight={"Regular"}>
        {I18n.t("wallet.errors.GENERIC_ERROR_SUBTITLE")}
      </H4>
    ),
    error => <ErrorCodeCopyComponent error={error} />
  );
```

would return the `ErrorCodeCopyComponent` on `UNCOVERED` cases which by design should have the GENERIC_ERROR_SUBTITLE text instead.

I'll give an attempt about these and propose another refactoring to you, in the next commit

File: ts/screens/wallet/payment/TransactionErrorScreen.tsx
Comment (CrisTofani) [reply ts/screens/wallet/payment/TransactionErrorScreen.tsx]: b3525ce to solve this

File: ts/screens/wallet/payment/TransactionErrorScreen.tsx
Comment (CrisTofani) [reply ts/screens/wallet/payment/TransactionErrorScreen.tsx]: f75e913 to solve

File: ts/screens/wallet/payment/TransactionErrorScreen.tsx
Comment (Undermaken) [inline ts/screens/wallet/payment/TransactionErrorScreen.tsx:121]: This PR LGTM
This is the only suggestion that I already tested (if you want to commit it directly)
```suggestion
    <H4 weight={"Bold"} testID={"error-code"}  style={{textAlign: "center"}}>
```

| now  | suggestion |
| ------------- | ------------- |
now | post

**Review by Undermaken — CHANGES_REQUESTED** [review]

**Review by Undermaken — APPROVED** [review]

**Comment by Undermaken** [comment]
@CrisTofani Waiting green light from CI when it comes back

**Comment by Undermaken** [comment]
@thisisjp for me is 🟢 
waiting your approval

**Review by thisisjp — APPROVED** [review]

**Pull request merged**
